src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -666,6 +701,68 @@ static void Main()
         }
 
         [Fact]
+        public void TupleWithDuplicateMemberNames()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int a, string a) x = (b: 1, c: ""hello"", b: 2);
+    }
+}
+" + trivial2uple + trivial3uple;
+
+            // PROTOTYPE we also expect duplicate member name error on the tuple type declaration
+            var comp = CreateCompilationWithMscorlib(source);
+            comp.VerifyDiagnostics(
+                // (6,50): error CS8202: Tuple member names must be unique.
+                //         (int a, string a) x = (b: 1, c: "hello", b: 2);
+                Diagnostic(ErrorCode.ERR_TupleDuplicateMemberName, "b").WithLocation(6, 50));
+        }
+
+        [Fact]
+        public void TupleWithReservedMemberNames()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int Item1, string Item3) x = (Item2: 1, Item4: ""hello"", Item3: 2);
+    }
+}
+" + trivial2uple + trivial3uple;
+
+            // PROTOTYPE we also expect reserved member name error on the tuple type declaration
+            var comp = CreateCompilationWithMscorlib(source);
+            comp.VerifyDiagnostics(
+                // (6,40): error CS8201: Tuple member name is disallowed at this position.
+                //         (int Item1, string Item3) x = (Item2: 1, Item4: "hello", Item3: 2);
+                Diagnostic(ErrorCode.ERR_TupleReservedMemberName, "Item2").WithLocation(6, 40),
+                // (6,50): error CS8201: Tuple member name is disallowed at this position.
+                //         (int Item1, string Item3) x = (Item2: 1, Item4: "hello", Item3: 2);
+                Diagnostic(ErrorCode.ERR_TupleReservedMemberName, "Item4").WithLocation(6, 50));
+        }
+
+        // PROTOTYPE this test can be removed once tuple-8 and above are implemented
+        [Fact]
+        public void LongTupleDoesntCrash()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        (int, int, int, int, int, int, int, int, int, int, int, int) x = (1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1);
+    }
+}
+" + trivial2uple;
+
+            CreateCompilationWithMscorlib(source).VerifyDiagnostics();
```

Comment:
We were talking about adding generic ERR_PrototypeNotYetImplemented error. I think it would work just fine for this case.